### PR TITLE
Add Test Cases for memcmp

### LINKED ISSUE #1860: Add unit test for memcmp()
Implement some test cases for memcmp()

### PR DESCRIPTION
**Pull request by sha-ahammed**
Closes #1860

### CODE REVIEW

File: src/tests/stdlib/memcmp_test.c [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * buf5 = buf4
+ */
+
+char buf1[] = "SWgaOtP43df2";
+char buf2[] = "SWGaOTP12df0";
+char buf3[] = "Hello";
+char buf4[] = "Welcome";
+char buf5[] = "Welcome";
+
+int n;
```
Comment (anton-bondarev) [inline src/tests/stdlib/memcmp_test.c]: don't use global variables

File: src/tests/stdlib/memcmp_test.c [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ EMBOX_TEST_SUITE("Test suite for memcmp()");
+
+ /**
+ * Test arrays
+ * buf1 > buf2
+ * buf4 > buf3
+ * buf5 = buf4
+ */
+
+char buf1[] = "SWgaOtP43df2";
```
Comment (anton-bondarev) [inline src/tests/stdlib/memcmp_test.c]: move all test string to TEST_CASEs

File: platform/pjsip/templates/x86-qemu/mods.config [hunk 3]
```diff
@@ -75,6 +75,7 @@ configuration conf {
 	@Runlevel(1) include embox.test.posix.ppty_test
 	@Runlevel(1) include embox.test.stdlib.bsearch_test
 	@Runlevel(1) include embox.test.stdlib.qsort_test
+	@Runlevel(1) include embox.test.stdlib.memcmp_test
```
Comment (anton-bondarev) [inline platform/pjsip/templates/x86-qemu/mods.config]: Rm it from here

File: src/tests/stdlib/memcmp_test.c [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+char buf2[] = "SWGaOTP12df0";
+char buf3[] = "Hello";
+char buf4[] = "Welcome";
+char buf5[] = "Welcome";
+
+int n;
+
+TEST_CASE("Test that buf1 > buf2")
```
Comment (anton-bondarev) [inline src/tests/stdlib/memcmp_test.c]: code-style
```TEST_CASE("fff") {```

File: src/tests/stdlib/memcmp_test.c [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+{
+   n = memcmp(buf1,buf2,sizeof(buf1));
+  test_assert(n > 0)
+}
+
+TEST_CASE("Test that buf3 < buf4")
+{
+   n = memcmp(buf3,buf4,sizeof(buf3));
```
Comment (anton-bondarev) [inline src/tests/stdlib/memcmp_test.c]: code-style
use only tabs at the start of line

**Comment by anton-bondarev** [comment]
@sha-ahammed Thanks
But you should be more careful with code style

File: src/tests/stdlib/memcmp_test.c [hunk 6]
```diff
@@ -11,35 +11,29 @@
...
-TEST_CASE("Test that buf1 > buf2")
-{
-   n = memcmp(buf1,buf2,sizeof(buf1));
-  test_assert(n > 0)
+TEST_CASE("Test that buf1 > buf2") {
+    int n;
+    char buf1[] = "SWgaOtP43df2";
+    char buf2[] = "SWGaOTP12df0";
+ 
+    n = memcmp(buf1,buf2,sizeof(buf1));
```
Comment (anton-bondarev) [inline src/tests/stdlib/memcmp_test.c]: code-style
`memcmp(buf1,buf2,sizeof(buf1));` -> `memcmp(buf1, buf2, sizeof(buf1));`

File: src/tests/stdlib/memcmp_test.c [hunk 7]
```diff
@@ -11,35 +11,29 @@
...
+    test_assert(n > 0)
 }
 
-TEST_CASE("Test that buf3 < buf4")
-{
-   n = memcmp(buf3,buf4,sizeof(buf3));
-  test_assert(n < 0);
+TEST_CASE("Test that buf3 < buf4") {
+    int n;
+    char buf3[] = "Hello";
```
Comment (anton-bondarev) [inline src/tests/stdlib/memcmp_test.c]: buf3 is a odd name

File: platform/pjsip/templates/x86-qemu/mods.config (hunk 3 above)
Comment (sha-ahammed) [reply platform/pjsip/templates/x86-qemu/mods.config]: Hey @anton-bondarev  I've updated the code to comply with code style and removed global variables.Sorry for mistake, i'm new to this.What does rm mean? i haven't figured out how to run tests yet.Thank you for your feedback.

File: src/tests/stdlib/memcmp_test.c (hunk 6 above)
Comment (sha-ahammed) [reply src/tests/stdlib/memcmp_test.c]: oops sorry

File: src/tests/stdlib/memcmp_test.c [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    char buf1[] = "SWgaOtP43df2";
+    char buf2[] = "SWGaOTP12df0";
+ 
+    n = memcmp(buf1, buf2, sizeof(buf1));
+    test_assert(n > 0);
+}
+
+TEST_CASE("Test that buf2 < buf4") {
+    int n;
+    char buf2[] = "Hello";
```
Comment (anton-bondarev) [inline src/tests/stdlib/memcmp_test.c]: I mean that you cat use only buf1 & buf2 for all test as I see

File: src/tests/stdlib/memcmp_test.c [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    char buf4[] = "Welcome";
+ 
+    n = memcmp(buf2, buf4, sizeof(buf2));
+    test_assert(n < 0);
+}
+
+TEST_CASE("Test's that two arrays are equal") {
+    int n;
+    char buf4[] = "Welcome";
```
Comment (anton-bondarev) [inline src/tests/stdlib/memcmp_test.c]: I mean buf4,5,6 looks strange

File: src/tests/stdlib/memcmp_test.c [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+
+ EMBOX_TEST_SUITE("Test suite for memcmp()");
+
+TEST_CASE("Test that buf1 > buf2") {
+    int n;
+    char buf1[] = "SWgaOtP43df2";
+    char buf2[] = "SWGaOTP12df0";
+ 
+    n = memcmp(buf1, buf2, sizeof(buf1));
+    test_assert(n > 0);
```
Comment (anton-bondarev) [inline src/tests/stdlib/memcmp_test.c]: test_assert_equal?

File: src/tests/stdlib/memcmp_test.c [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+ * @date March 18,2020
+ * @author Sha Ahammed Roze
+ */
+
+ #include <embox/test.h>
+ #include <strings.h>
+
+ EMBOX_TEST_SUITE("Test suite for memcmp()");
+
+TEST_CASE("Test that buf1 > buf2") {
```
Comment (anton-bondarev) [inline src/tests/stdlib/memcmp_test.c]: Test description has not enough information
"Test that  SWgaOtP43df2 > SWGaOTP12df0"

**Comment by anton-bondarev** [comment]
@sha-ahammed Can you add some more tests?
For example checking passing NULL arguments?

File: src/tests/stdlib/memcmp_test.c (hunk 9 above)
Comment (sha-ahammed) [reply src/tests/stdlib/memcmp_test.c]: oh only use 2 names? buf1 and buf 2?

File: src/tests/stdlib/memcmp_test.c (hunk 10 above)
Comment (sha-ahammed) [reply src/tests/stdlib/memcmp_test.c]: test_assert_equal should be true? I though it automatically checked that?

**Comment by sha-ahammed** [comment]
@anton-bondarev yeah i can add more tests. First let me address these changes first.

File: src/tests/stdlib/memcmp_test.c (hunk 10 above)
Comment (sha-ahammed) [reply src/tests/stdlib/memcmp_test.c]: @anton-bondarev doesnt test pass if the condition inside test_asset is true? why do i need test_assert_equal?

Comment (anton-bondarev) [reply src/tests/stdlib/memcmp_test.c]: @sha-ahammed 
I mean you can use strict equality
test_assert(n > 0); -> test_assert(n == ..);

Comment (anton-bondarev) [reply src/tests/stdlib/memcmp_test.c]: ok, it's good

**Comment by anton-bondarev** [comment]
@sha-ahammed That tests are ok
Add some more tests please

**Comment by sha-ahammed** [comment]
@anton-bondarev  isnt passing null arguments undefined for memcmp?

**Comment by anton-bondarev** [comment]
> @anton-bondarev isnt passing null arguments undefined for memcmp?

NULL arguments it is only the example. Another is there any test comparing only part of buffers, or call memcmp with zero size

**Comment by sha-ahammed** [comment]
@anton-bondarev  ive added 4 more tests, can your review?

**Comment by anton-bondarev** [comment]
@sha-ahammed Thanks. It's enough for the first issue.
Please rm your changes (add commits with removing) from templates
files:
 * platform/pjsip/templates/x86-qemu/mods.config
 * templates/x86/qemu/mods.config

**Comment by sha-ahammed** [comment]
@anton-bondarev  did i do it right?

**Comment by anton-bondarev** [comment]
> @anton-bondarev did i do it right?

Yes, thanks.
Now rebase under embox/master please.
Be careful there are some conflict.

**Comment by sha-ahammed** [comment]
@anton-bondarev i resolved conflict.

**Comment by sha-ahammed** [comment]
@anton-bondarev  i think i made a mistake somewhere.

**Comment by anton-bondarev** [comment]
> @anton-bondarev i think i made a mistake somewhere.

Don't worry. Let's try again :)
```
git fetch upstream
git rebase upstream/master
```
if there are some conflicts fix them
then as you use 'master' branch (it's not very good) but
```
git push origin +master
```
```

**Comment by sha-ahammed** [comment]
@anton-bondarev  is it ok to merge instead of rebase?

**Comment by anton-bondarev** [comment]
> @anton-bondarev is it ok to merge instead of rebase?

No, unfortunately not :(

You can create a new PR
copy your changes to a new branch
```
git checkout -b my_new_branch
```
push the new branch to your origin
```
git push -u origin my_new_branch
```
and then try rebase again

**Comment by anton-bondarev** [comment]
@sha-ahammed Are there any troubles with the new PR creation?
